core/templates/dev/head/pages/exploration_player/PlayerServices.js

```diff
@@ -122,6 +128,9 @@ oppia.factory('oppiaPlayerService', [
       if (!_editorPreviewMode) {
         StatsReportingService.recordExplorationStarted(
           exploration.initStateName, newParams);
+        totalTimeStopwatch.reset();
+        $rootScope.$broadcast(
+          'sessionTime', totalTimeStopwatch.getTimeInSecs(), initialState.name);
```

Comment:
I'm a bit unclear about what exactly sessionTime is supposed to mean -- the name isn't very clear. Is it the "start time"? If not, maybe call it startTime or sessionStartTime?

Also I looked at what was receiving the $broadcast and it just seems to be another function in this service. If that's the case why not just inline it here?

Reply:
@seanlip Are you suggesting that we check session time in each of the places we broadcast an event? That way, we wouldn't need to use $broadcast at all.
Oh, and sessionTime is supposed to be the total time since the exploration started. The stopwatch currently resets in every state transition, whereas session transition doesn't.

Reply:
OK, I'm fine with having a different "entire exploration" stopwatch.

No, I'm just saying, you only have a single $on('sessionTime') receiver, AFAICT. So, why not just extract that into a function and call that function in the places where you're currently doing a $broadcast?

Reply:
Makes sense. Done.